app/components/ScrollToButton.tsx

```diff
@@ -20,9 +21,20 @@ export function useScrollToEvent(scheduleIndex) {
   const currentEventIndex = useSharedValue(-1)
   const currentlyViewingEvents = useSharedValue<number[]>([])
   const currentlyViewingSchedule = useSharedValue(0)
+  const isFocused = useSharedValue(false)
+
+  useFocusEffect(
```

Comment:
I think you could combine this and the useSharedValue line to 

`const is focused = useIsFocused()`

Check ScheduleScreen.tsx for an example

Reply:
That doesn't seem to work. If I'm understanding right it's because `useRef` doesn't take dependencies so the callback closes over the initial value of `isFocused` from `useIsFocused()` and never updates again. Unfortunately the callback `onViewableItemsChanged` needs to be wrapped in a ref or it throws errors so I don't think there's a way around this. I did try to turn the callback into a higher order function so I could pass `isFocused` from the ScheduleScreen in but that didn't work either. I think that's due to similar reasons but I'm less sure why that didn't work. It seems like once you start using shared values it's easier to just use shared values (e.g. like how I fixed the arrow flickering).

Reply:
I've also had issues with `useIsFocused`.

Reply:
Interesting OK good to know